frontend/src/lib/components/Cards/InsightCard/InsightDetails.tsx

```diff
@@ -329,23 +329,30 @@ function InsightDetailsInternal({ insight }: { insight: InsightModel }, ref: Rea
             <div className="InsightDetails__footer">
                 <div>
                     <h5>Created by</h5>
-                    <section>
-                        <ProfilePicture name={created_by?.first_name} email={created_by?.email} showName size="md" />{' '}
-                        <TZLabel time={created_at} />
-                    </section>
-                </div>
-                <div>
-                    <h5>Last modified by</h5>
                     <section>
                         <ProfilePicture
-                            name={insight.last_modified_by?.first_name}
-                            email={insight.last_modified_by?.email}
+                            name={created_by?.first_name || 'PostHog via feature flag'}
```

Comment:
Hey @ahsan722505 , this isn't really sufficient here, since something else might also generate insights. Turn this into a generic "Generated by PostHog" instead.

(Team-product-analytics, are there any other cases where created_by might not exist? like if the user left/got deleted? And is there a way for us to figure this out?)

```suggestion
                            name={created_by?.first_name || 'Generated by PostHog'}
```

Reply:
It's possible for a user to be null if they've been deleted due to a GDPR request. This is why I'm a bit wary of treating null as "generated by PostHog". I think some sort of instance-wide "PostHog System" user would be good for marking auto-generated things.

Reply:
@neilkakkar @Twixes Does it look good?
![image](https://github.com/PostHog/posthog/assets/79001785/0b0a50dc-56dc-423a-b63f-3309d0700013)

Reply:
I'm afraid this doesn't address the issue, because if a user is deleted from our system, then the insight would show up as "created by PostHog", which would be false.

Reply:
Can you recommend a solution that would cover all the use cases?

Reply:
The consistent solution will be more involved than just a frontend change, but the issue is that we cannot treat a null user value as a "PostHog system" user. Null must literally mean "unknown user".
I think what we can do is: we can ensure there's a "PostHog System" user at server start, that user having a special ID like 0 or -1, which we'll then be able to use in `created_by_id` columns. `UserSerializer` will need to have special handling for that user, since it'll be read-only and not part of any PostHog organization. This way both foreign key constraints, data correctness, and UI clarity would be satisfied.